src/passes.rs

```diff
@@ -920,12 +935,41 @@ impl<T: PassManagerSubType> PassManager<T> {
     ///
     /// NOTE: this pass has a habit of making definitions be dead. It is a good idea to
     /// run a Dead Instruction Elimination pass sometime after running this pass.
+    ///
+    /// In LLVM 12 and later, this instruction is replaced by the
+    /// [`add_instruction_simplify_pass`].
+    #[llvm_versions(3.6..=11.0)]
+    #[doc(alias = "add_instruction_simplify_pass")]
     pub fn add_constant_propagation_pass(&self) {
         unsafe {
             LLVMAddConstantPropagationPass(self.pass_manager)
         }
     }
 
+    /// This pass implements constant propagation and merging. It looks for instructions
+    /// involving only constant operands and replaces them with a constant value instead
+    /// of an instruction. For example:
+    ///
+    /// ```ir
+    /// add i32 1, 2
+    /// ```
+    ///
+    /// becomes
+    ///
+    /// ```ir
+    /// i32 3
+    /// ```
+    ///
+    /// NOTE: this pass has a habit of making definitions be dead. It is a good idea to
+    /// run a Dead Instruction Elimination pass sometime after running this pass.
```

Comment:
based on the LLVM docs/source this comment is still accurate

Reply:
Cool, ty for details docs!